docs/SCTValidation.md

```diff
@@ -0,0 +1,259 @@
+verifying the submitted certificate chain.  The chain of signatures has to be
+verified using the full chain, including the pre-issuer, but other validity
+checks need to be done as if the pre-issuer were not present.  For example:
+ - The pre-issuer has the CT EKU, but the leaf cert does not, so any check that
+   the leaf's EKUs are a subset of its issuer's EKUs will incorrectly fail.
+ - Any path length constraint (in a
+   [Basic Constraints](https://tools.ietf.org/html/rfc5280#section-4.2.1.9)
+   extension) for a CA certificate in the chain may be off by one (and so
+   theoretically not allow the pre-issuer to be a CA certificate).
+
+So to sum up *pre-issued* precertificates:
+ - The CA builds a precertificate version of the certificate that includes the
+   poison.
+ - Next, the CA modifies the Issuer and (optional) Authority Key Identifier
+   extension in the precertificate to match the pre-issuer rather than the true
+   issuer.
+ - The CA signs the resulting precertificate using the pre-issuer's key.
+ - The submits the whole chain (precert, pre-issuer, true-issuer, ... root) to
+   the Log.
+ - The Log removes the poison and extracts the inner `tbsCertificate`.
+ - The Log notices that the direct issuer has the Certificate Transparency
+   extended key usage, so treats the next entry in the chain as the true issuer.
+ - The Log updates the Issuer and (optional) Authority Key Identifier extension
+   to match the true issuer rather than the pre-issuer.
+ - The Log combines the resulting inner `tbsCertificate` with the hash of the
+   (true) issuer's key, to form the core `PreCert` data that the Log deals with.
+ - The Log sends an SCT back to the CA, which includes a timestamp as usual but
+   has a signature over data including the `PreCert` bundle of `tbsCertificate`
+   and issuer key hash.
+ - The CA builds an SCT list extension that includes this SCT, attaches it to
+   the original (un-poisoned) certificate, and signs the whole thing with the
+   true issuer's key.
+
+This is complicated, but it's worth pointing out that only the CA and the Log
+need to deal with all of this; the SCT (and the tree leaf it corresponds to)
+only covers data that pertains to the final certificate.  This means that a
+client who receives a certificate with an embedded SCT can just do the
+[same checks](#embedded-scts) as for any other embedded SCT.
+
+
+## SCT Validation Steps
+
+There are two main aspects to validating an SCT:
+ - Signature Validation: this requires the following (in addition to the SCT itself):
+     - the Log's public key
+     - the [pre]certificate data that the signature encompasses
+     - (for a precertificate) the issuer's public key
+ - Inclusion Checking: this requires the following (in addition to the SCT
+   itself):
+     - enough time to have passed (the MMD) for the certificate to be incorporated
+     - the Log's URL (for accessing the `get-proof-by-hash`
```

Comment:
and the Log's public key :)

Reply:
Added, although for inclusion it's only used for checking the STH.